src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2057,6 +2103,19 @@ private bool HasArrayConversionToInterface(ArrayTypeSymbol source, TypeSymbol de
             return HasIdentityOrImplicitReferenceConversion(source.ElementType.TypeSymbol, argument0, ref useSiteDiagnostics);
         }
 
+        private bool HasImplicitReferenceConversion(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            if (IncludeNullability)
+            {
+                if (!HasTopLevelNullabilityImplicitConversion(source, destination))
+                {
+                    return false;
+                }
+                return HasIdentityOrImplicitReferenceConversion(source.TypeSymbol, destination.TypeSymbol, ref useSiteDiagnostics);
```

Comment:
>return HasIdentityOrImplicitReferenceConversion(source.TypeSymbol, destination.TypeSymbol, ref useSiteDiagnostics); [](start = 16, length = 115)

It feels confusing that in ```IncludeNullability``` mode we allow identity conversions in this helper. It is hard to infer the impact of this. I think it would be better to explicitly check for identity where it should be done rather than relying on a side-effect of this helper.